assets/src/edit-story/components/library/panes/media/mediaPane.js

```diff
@@ -183,8 +183,19 @@ function MediaPane(props) {
    * @return {null|*} Return onInsert or null.
    */
   const insertMediaElement = (resource) => {
-    const width = Math.min(resource.width * DEFAULT_DPR, DEFAULT_ELEMENT_WIDTH);
-    return insertElement(resource.type, { resource, width });
+    const newWidth =
+      resource.posterGenerated && resource.posterWidth
+        ? resource.posterWidth
+        : resource.width;
+    const newHeight =
+      resource.posterGenerated && resource.posterHeight
+        ? resource.posterHeight
+        : resource.height;
```

Comment:
Perhaps we could use something like
`const hasGeneratedPoster = resource.posterGenerated && resource.posterWidth && resource.posterHeight` and then use that as the condition instead? To ensure that poster measures are used only if both width & height exist.

Reply:
Yeah, that would tidy up the code. Let me push that change.

Reply:
Fixed in https://github.com/google/web-stories-wp/pull/1102/commits/9f507e30edb52d6b3ae4bb1d755c16561b3cf259